contracts/TransmuteDPOS.sol

```diff
@@ -86,22 +85,21 @@ contract TransmuteDPOS is TransmuteToken, RoundManager, ProviderPool {
     Delegator storage d = delegators[msg.sender];
     // Provider has to be a Delegator to himself
     require(d.delegateAddress == msg.sender);
-    if (p.status == ProviderStatus.Unregistered) {
+    if (providerStatus(msg.sender) == ProviderStatus.Unregistered) {
       numberOfProviders = numberOfProviders.add(1);
       addProvider(msg.sender, p.totalAmountBonded);
       emit ProviderAdded(msg.sender, _pricePerStorageMineral, _pricePerComputeMineral, _blockRewardCut, _feeShare);
     } else {
       emit ProviderUpdated(msg.sender, _pricePerStorageMineral, _pricePerComputeMineral, _blockRewardCut, _feeShare);
     }
-    p.status = ProviderStatus.Registered;
     p.pricePerStorageMineral = _pricePerStorageMineral;
     p.pricePerComputeMineral = _pricePerComputeMineral;
     p.blockRewardCut = _blockRewardCut;
     p.feeShare = _feeShare;
   }
 
   function resignAsProvider(address _provider) internal {
-    require(providers[_provider].status != ProviderStatus.Unregistered);
+    require(providerStatus(_provider) != ProviderStatus.Unregistered);
```

Comment:
I feel like this should check to ensure that the provider is `Registered` vs not `Unregistered`, for readability purposes.